src/main/java/by/gsu/duelingobackend/controller/RelationshipController.java

```diff
@@ -0,0 +1,58 @@
+package by.gsu.duelingobackend.controller;
+
+import by.gsu.duelingobackend.dto.request.RelationshipRequest;
+import by.gsu.duelingobackend.dto.response.RelationshipResponse;
+import by.gsu.duelingobackend.service.RelationshipService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/relationships")
+@RequiredArgsConstructor
+public class RelationshipController {
+
+    private final RelationshipService relationshipService;
+
+    @PostMapping("/friend-requests")
+    @ResponseStatus(HttpStatus.CREATED)
+    public RelationshipResponse sendFriendRequest(@RequestBody RelationshipRequest request) {
+        return relationshipService.sendFriendRequest(request);
+    }
+
+    @GetMapping("/friend-requests/incoming")
+    public List<RelationshipResponse> getIncomingRequests(@RequestParam UUID userId) {
+        return relationshipService.getIncomingRequests(userId);
+    }
+
+    @PatchMapping("/friend-requests/{requestId}")
+    public RelationshipResponse updateRelationshipStatus(
+            @PathVariable UUID requestId,
+            @RequestParam String action
+    ) {
+        return relationshipService.updateRelationshipStatus(requestId, action);
+    }
+
+    @PostMapping("/blocks")
+    public RelationshipResponse blockUser(@RequestBody RelationshipRequest request) {
```

Comment:
I think those 2 methods should be in user controller, also why we need all request for block user?